src/Interpreters/ProcessList.h

```diff
@@ -301,11 +306,11 @@ class ProcessList
     size_t max_select_queries_amount = 0;
 
     /// amount of queries by query kind.
-    QueryKindToAmount query_kind_amounts;
+    QueryKindAmounts * query_kind_amounts = new QueryKindAmounts{0, 0};
```

Comment:
Why we changed generic solution with map, to solution that support only 2 query kind types ? Even if this class works with only insert, select query kinds it is easier to have generic solution, as client I do not except that `increaseQueryKindAmount` will not work for any query kind.

Reply:
I get the map back here